src/query/fuzzy_query.rs

```diff
@@ -129,30 +110,39 @@ impl FuzzyTermQuery {
     }
 
     fn specialized_weight(&self) -> crate::Result<AutomatonWeight<DfaWrapper>> {
-        // LEV_BUILDER is a HashMap, whose `get` method returns an Option
-        match LEV_BUILDER.get(&(self.distance, self.transposition_cost_one)) {
-            // Unwrap the option and build the Ok(AutomatonWeight)
-            Some(automaton_builder) => {
-                let term_text = self.term.as_str().ok_or_else(|| {
-                    crate::TantivyError::InvalidArgument(
-                        "The fuzzy term query requires a string term.".to_string(),
-                    )
-                })?;
-                let automaton = if self.prefix {
-                    automaton_builder.build_prefix_dfa(term_text)
-                } else {
-                    automaton_builder.build_dfa(term_text)
-                };
-                Ok(AutomatonWeight::new(
-                    self.term.field(),
-                    DfaWrapper(automaton),
+        static AUTOMATON_BUILDER: [[OnceCell<LevenshteinAutomatonBuilder>; 2]; 3] = [
+            [OnceCell::new(), OnceCell::new()],
+            [OnceCell::new(), OnceCell::new()],
+            [OnceCell::new(), OnceCell::new()],
+        ];
+
+        let automaton_builder = AUTOMATON_BUILDER
+            .get(self.distance as usize)
+            .ok_or_else(|| {
+                InvalidArgument(format!(
+                    "Levenshtein distance of {} is not allowed. Choose a value less than {}",
+                    self.distance,
+                    AUTOMATON_BUILDER.len()
                 ))
-            }
-            None => Err(InvalidArgument(format!(
-                "Levenshtein distance of {} is not allowed. Choose a value in the {:?} range",
-                self.distance, VALID_LEVENSHTEIN_DISTANCE_RANGE
-            ))),
-        }
+            })?
+            .get(self.transposition_cost_one as usize)
+            .unwrap()
```

Comment:
```suggestion
            [self.transposition_cost_one as usize]
```

Reply:
would not that be the same?

Reply:
It would certainly be the same, but it formats very unfavorably:

```rust
let automaton_builder = AUTOMATON_BUILDER
    .get(self.distance as usize)
    .ok_or_else(|| {
        InvalidArgument(format!(
            "Levenshtein distance of {} is not allowed. Choose a value less than {}",
            self.distance,
            AUTOMATON_BUILDER.len()
        ))
    })?[self.transposition_cost_one as usize]
    .get_or_init(|| {
        LevenshteinAutomatonBuilder::new(self.distance, self.transposition_cost_one)
    });
```

which is why I went for this somewhat verbose way of doing it. But I am not really invested in this.

Which one do you prefer after seeing the formatting, `[..]` or `get(..).unwrap()`?